packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -740,7 +740,7 @@ export async function signTransaction<ReturnFormat extends DataFormat>(
 ) {
 	const response = await ethRpcMethods.signTransaction(
 		web3Context.requestManager,
-		formatTransaction(transaction, ETH_DATA_FORMAT),
+		formatTransaction(transaction as TransactionWithSenderAPI, ETH_DATA_FORMAT),
```

Comment:
we dnt need to cast it here, as issue under discussion was for `getTransaction` and other functions in this file @avkos

Reply:
reverted